WordPress/Classes/ViewRelated/Blog/BlogDetailHeaderView.m

```diff
@@ -209,4 +214,44 @@ - (void)setUpdatingIcon:(BOOL)updatingIcon
     }
 }
 
+#pragma mark - Drop Interaction Handler
+- (BOOL)dropInteraction:(UIDropInteraction *)interaction
+       canHandleSession:(id<UIDropSession>)session API_AVAILABLE(ios(11.0))
+{
+    BOOL isAnImage = [session canLoadObjectsOfClass:[UIImage self]];
+    BOOL isSingleImage = [session.items count] == 1;
+    return (isAnImage && isSingleImage);
+}
+
+- (UIDropProposal *)dropInteraction:(UIDropInteraction *)interaction
+                   sessionDidUpdate:(id<UIDropSession>)session API_AVAILABLE(ios(11.0))
+{
+    CGPoint dropLocation = [session locationInView:self];
+    UIDropOperation dropOperation;
+    
+    if (CGRectContainsPoint(self.blavatarImageView.frame, dropLocation)) {
+        dropOperation = UIDropOperationCopy;
+    } else {
+        dropOperation = UIDropOperationCancel;
+    }
+    
+    UIDropProposal *dropProposal = [[UIDropProposal alloc] initWithDropOperation:dropOperation];
+    
+    return  dropProposal;
+}
+
+- (void)dropInteraction:(UIDropInteraction *)interaction
+            performDrop:(id<UIDropSession>)session API_AVAILABLE(ios(11.0))
+{
+    [self setUpdatingIcon:YES];
+    
+    __weak __typeof(self) weakSelf = self;
+    [session loadObjectsOfClass:[UIImage self] completion:^(NSArray *images) {
+        UIImage *image = [images firstObject];
+        weakSelf.blavatarImageView.image = image;
```

Comment:
We could probably hold off on updating the image in the image view until the user's cropped / confirmed it?